### PR TITLE
[BST-263] feature: dashboard page

### PR DESCRIPTION
**Pull request by mihaimdm22**

### CODE REVIEW

File: pages/dashboards/[dashboardId].tsx [hunk 1]
```diff
@@ -0,0 +1,50 @@
+import { Button, ButtonGroup, Link } from "@chakra-ui/react";
+import { PencilAltIcon } from "@heroicons/react/outline";
+import { useACLHelpers } from "@/features/authorization/hooks";
+import { useDashboardResponse } from "@/features/dashboards/hooks";
+import { useDataSourceContext } from "@/hooks";
+import { useDataSourceResponse } from "@/features/data-sources/hooks";
+import Layout from "@/components/Layout";
+import PageWrapper from "@/components/PageWrapper";
+
+const ViewShow = () => {
```
Comment (adrianthedev) [inline pages/dashboards/[dashboardId].tsx:10]: Can rename this to `DashboardView` or something better?

File: pages/dashboards/[dashboardId].tsx [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    <Layout>
+      <PageWrapper
+        heading={`Dashboard ${dashboard?.name}`}
+        flush={true}
+        buttons={
+          <ButtonGroup size="xs">
+            {canEdit && dashboardId && <EditDashboardButton />}
+          </ButtonGroup>
+        }
+        footer={<></>}
```
Comment (adrianthedev) [inline pages/dashboards/[dashboardId].tsx:40]:
```suggestion
```

File: pages/dashboards/[dashboardId].tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+        Edit dashboard
+      </Button>
+    </Link>
+  );
+
+  return (
+    <Layout>
+      <PageWrapper
+        heading={`Dashboard ${dashboard?.name}`}
+        flush={true}
```
Comment (adrianthedev) [inline pages/dashboards/[dashboardId].tsx:34]: Maybe we don't need it flush?
```suggestion
```

File: pages/dashboards/[dashboardId]/edit.tsx [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+              Delete dashboard
+            </Button>
+          ),
+        }}
+        buttons={
+          dashboardId && (
+            <BackButton href={backLink}>Back to dashboard</BackButton>
+          )
+        }
+        flush={true}
```
Comment (adrianthedev) [inline pages/dashboards/[dashboardId]/edit.tsx:105]: Same here?
```suggestion
```

File: pages/dashboards/[dashboardId]/edit.tsx [hunk 5]
```diff
@@ -0,0 +1,128 @@
...
+  );
+
+  const [removeDashboard, { isLoading: dashboardIsRemoving }] =
+    useRemoveDashboardMutation();
+
+  const handleRemove = async () => {
+    if (dashboardIsLoading || dashboardIsRemoving) return;
+
+    const confirmed = confirm(
+      "Are you sure you want to remove this dashboard? All information about it (settings included) will be completely removed from our servers."
```
Comment (adrianthedev) [inline pages/dashboards/[dashboardId]/edit.tsx:40]:
```suggestion
      "Are you sure you want to remove this dashboard? All information, settings, and widgets will be completely removed from our servers."
```

File: pages/api/dashboards/index.ts [hunk 6]
```diff
@@ -0,0 +1,128 @@
...
+    get(user, ["organizations"])
+      .map((orgUser) => orgUser?.organization)
+      .map((org) => org?.dashboards)
+  );
+
+  res.json(ApiResponse.withData(dashboards));
+}
+
+export default withMiddlewares(handler, {
+  middlewares: [[IsSignedIn, {}]],
```
Comment (adrianthedev) [inline pages/api/dashboards/index.ts:127]: Maybe we should check if he has access to the dataSource?

Comment (adrianthedev) [reply pages/api/dashboards/index.ts:127]: We can also move the POST method to `/data-sources/[id]/dashboards`.

File: hooks/index.ts [hunk 7]
```diff
@@ -127,10 +128,20 @@ export const useDataSourceContext = () => {
 
   useEffect(() => {
     if (view?.dataSourceId)
-      dispatch(setDataSourceId(view.dataSourceId.toString()));
+    dispatch(setDataSourceId(view.dataSourceId.toString()));
```
Comment (adrianthedev) [inline hooks/index.ts:131]: This is not properly indented.

File: hooks/index.ts [hunk 8]
```diff
@@ -127,10 +128,20 @@ export const useDataSourceContext = () => {
...
     if (view?.tableName) dispatch(setTableName(view.tableName));
   }, [view]);
 
+  const dashboardId = router.query.dashboardId as string;
+  const { dashboard } = useDashboardResponse(dashboardId);
+
+  useEffect(() => {
+    if (dashboard?.dataSourceId) {
+      dispatch(setDataSourceId(dashboard.dataSourceId.toString()));
+      dispatch(setTableName(""));
```
Comment (adrianthedev) [inline hooks/index.ts:141]: Maybe we need to invalidate the view too?

**Review by adrianthedev — CHANGES_REQUESTED** [review]
GG. Just a few things to take care of

- [x] Can we add the default "create dashboard" box like on views?
- [x] Check the permissions. We don't want data leaks

image

File: hooks/index.ts (hunk 8 above)
Comment (mihaimdm22) [reply hooks/index.ts:141]: not necessary

**Pull request merged**